Source/CesiumRuntime/Private/Cesium3DTileset.cpp

```diff
@@ -1635,18 +1635,24 @@ void removeCollisionForTiles(
 void applyActorCollisionSettings(
     const FBodyInstance& BodyInstance,
     UCesiumGltfComponent* Gltf) {
-  UCesiumGltfPrimitiveComponent* PrimitiveComponent =
-      static_cast<UCesiumGltfPrimitiveComponent*>(Gltf->GetChildComponent(0));
-  if (PrimitiveComponent != nullptr) {
-    if (PrimitiveComponent->GetCollisionObjectType() !=
-        BodyInstance.GetObjectType()) {
-      PrimitiveComponent->SetCollisionObjectType(BodyInstance.GetObjectType());
-    }
-    const UEnum* ChannelEnum = StaticEnum<ECollisionChannel>();
-    if (ChannelEnum) {
-      FCollisionResponseContainer responseContainer =
-          BodyInstance.GetResponseToChannels();
-      PrimitiveComponent->SetCollisionResponseToChannels(responseContainer);
+  const TArray<USceneComponent*>& ChildrenComponents =
+      Gltf->GetAttachChildren();
+
+  for (USceneComponent* ChildComponent : ChildrenComponents) {
+    UCesiumGltfPrimitiveComponent* PrimitiveComponent =
+        static_cast<UCesiumGltfPrimitiveComponent*>(ChildComponent);
```

Comment:
static_cast'ing a `USceneComponent` to a `UCesiumGltfPrimitiveComponent` when it is not actually a `UCesiumGltfPrimitiveComponent` will result in undefined behavior. static_cast isn't smart enough to return nullptr in this scenario. In regular C++ you would need `dynamic_cast`, but in Unreal we should use Unreal's `Cast` template.

Reply:
The original code has this problem, too. 😆 We should fix it while we're in here, though.